### PR TITLE
created a general page for mindsdb handlers

### PR DESCRIPTION
**Pull request by martyna-mindsdb**
## Description

- created a general page for mindsdb handlers: 
[Handlers in MindsDB - MindsDB.pdf](https://github.com/mindsdb/mindsdb/files/10229597/Handlers.in.MindsDB.-.MindsDB.pdf)
- updated a page on database handlers: 
[Building a Database Handler - MindsDB.pdf](https://github.com/mindsdb/mindsdb/files/10229601/Building.a.Database.Handler.-.MindsDB.pdf)
- updated a page on ml handlers: 
[Building an ML Engine - MindsDB.pdf](https://github.com/mindsdb/mindsdb/files/10229604/Building.an.ML.Engine.-.MindsDB.pdf)

**Fixes** #4055

## Type of change

(Please delete options that are not relevant)

- [ ] ⚡ New feature (non-breaking change which adds functionality)

### What is the solution?

As in Description.

## Checklist:

- [ ] I have updated the documentation.

### CODE REVIEW

File: docs/contribute/integrations.mdx [hunk 1]
```diff
@@ -3,53 +3,33 @@ title: Building a Database Handler
 sidebarTitle: Building a Database Handler
 ---
 
-In this section, you'll find how to add new integrations to MindsDB, either as *data layers* or *machine learning (ML) frameworks*.
+In this section, you'll find how to add new integrations/databases to MindsDB.
```
Comment (paxcema) [inline docs/contribute/integrations.mdx:6]: I'd suggest to rename this file as `docs/contribute/data-handlers.mdx`

File: docs/mindsdb-handlers.mdx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+The entry point for a database handler is a class definition that should inherit directly from the [mindsdb.integrations.libs.base.DatabaseHandler](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L102) class and thus indirectly from the [mindsdb.integrations.libs.base.BaseHandler](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L8) class, which defines all the methods that must be overwritten in order to achieve a functional implementation.
+
+And the entry point for an ML handler is a class definition that should inherit from the [mindsdb.integrations.libs.base.BaseMLEngine](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L122) class, which defines all the methods that must be overwritten in order to achieve a functional implementation.
+
+<Note>
+    The handler's structure is not enforced, and the package design is up to the author.
```
Comment (paxcema) [inline docs/mindsdb-handlers.mdx:56]: I would rephrase slightly to:

> All other details of the handler's structure are not enforced, so the author can decide on its design.

File: docs/mindsdb-handlers.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+Inherit from the [`BaseMLEngine`](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/base.py#L122) class when adding a new ML handler. For more info, visit our [doc page here](/contribute/ml-handlers/).
+
+### Parsing SQL
+
+Whenever you want to parse a string that contains SQL, we strongly recommend using the `mindsdb_sql` package. It provides a parser that fully supports the MindsDB SQL dialect and partially the standard SQL dialect. There is also a *render* feature to map other dialects into the already supported ones.
+
+### Formatting Output
+
+When it comes to building the response of the public methods, the output should be wrapped by the [mindsdb.integrations.libs.response.HandlerResponse](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/response.py#L7) or [mindsdb.integrations.libs.response.HandlerStatusResponse](https://github.com/mindsdb/mindsdb/blob/staging/mindsdb/integrations/libs/response.py#L32) class. These classes are used by the MindsDB executioner to orchestrate and coordinate multiple handler instances in parallel.
```
Comment (paxcema) [inline docs/mindsdb-handlers.mdx:75]: Let's specify that this applies to the data handlers. For ML handlers, this is automatically done by an intermediate wrapper (`BaseMLEngineExec`) so the contributor wouldn't need to worry about it.

File: docs/contribute/ml-handlers.mdx [hunk 4]
```diff
@@ -6,58 +6,54 @@ sidebarTitle: Building an ML Engine
...
 | `create_engine()` | Optional. It connects with external sources, such as REST API.                       |
 
-### Implementing Mandatory Methods
+Authors can opt for adding private methods, new files and folders, or any combination of these to structure all the necessary work that will enable the core methods to work as intended.
+
+<Tip>
+    **Other Common Methods**
+    
+    Under the `mindsdb.integrations.libs.utils` library, contributors can find various methods that may be useful while implementing new handlers.
+</Tip>
```
Comment (paxcema) [inline docs/contribute/ml-handlers.mdx:50]: Similarly to the general intro page, here we should add a tip that mentions there is a wrapper for `BaseMLEngine`s called `BaseMLEngineExec` which is automatically deployed and takes care of modifying the data responses here into something that can be used alongside data handlers.

**Review by paxcema — CHANGES_REQUESTED** [review]

**Review by paxcema — APPROVED** [review]

**Review by ZoranPandovski — APPROVED** [review]
